frontend/src/metabase/querying/notebook/components/NotebookDataPicker/EmbeddingDataPicker/EmbeddingDataPicker.tsx

```diff
@@ -74,14 +90,22 @@ export function EmbeddingDataPicker({
     );
   }
 
+  const isSourceSelected = Boolean(pickerInfo?.tableId);
   return (
     <PLUGIN_EMBEDDING.DataSourceSelector
-      key={pickerInfo?.tableId}
-      isInitiallyOpen={!table}
+      key={
+        isSourceSelected
+          ? pickerInfo?.tableId
+          : `${sourceTable?.id}:${isSourceModelFetching}`
+      }
+      isInitiallyOpen={isSourceModelFetching ? false : !table}
+      isQuerySourceModel={isSourceModel}
       canChangeDatabase={canChangeDatabase}
       selectedDatabaseId={databaseId}
       selectedTableId={pickerInfo?.tableId}
-      selectedCollectionId={card?.collection_id}
+      selectedCollectionId={
+        normalizedCard?.collection_id ?? sourceModelCollectionId
```

Comment:
Just curious why we need `sourceModelCollectionId` here - is this because `normalizedCard` can be `undefined` as in `const normalizedCard = pickerInfo?.cardId ? card : undefined;`?

Reply:
This prop determines what collection to open when rendering the `SavedEntityPicker` or the model step.

We want the join step to open the same collection as the staring model (model at the first query stage). Here's [the ref](https://metaboat.slack.com/archives/C063Q3F1HPF/p1739808652233439?thread_ts=1739800920.795329&cid=C063Q3F1HPF).

`normalizedCard?.collection_id` is used when we've selected a model so clicking the data picker again should open the collection of the model we just selected.

`sourceModelCollectionId` is used when we're joining as I've mentioned before. It will show the collection of the source model.